### ISSUE TITLE
[Accessibility] Admin order "Add Note" button incorrectly coded as link

### ISSUE THREAD

**Issue by amberhinds** [woothemes/woocommerce#60964]
### Prerequisites

- [x] I have carried out troubleshooting steps and I believe I have found a bug.
- [x] I have searched for similar bugs in both open and closed issues and cannot find a duplicate.

### Describe the bug

The "Add Note" button is incorrectly coded as a link.

<img width="848" height="494" alt="Image" src="https://github.com/user-attachments/assets/7faf3bdb-bf69-4ca5-ba59-4ff1157dde6a" />

Relevant code:
`<a href="#" class="add_note button" data-id="147">Add</a>`

### Expected behavior

Buttons perform actions. Links go somewhere. As this submits the form, it should announce as a button to screen reader users and be able to be triggered with a space bar.

Use the appropriate `<button>` tag on the element.

If you cannot replace a `<a>` tag with a button, it is possible to remediate a link to function as a button with ARIA. The link would need to be given a button role, and you would also need to write JavaScript to ensure that the link will function properly as a button for keyboard users and can be triggered with the space bar in addition to the return/enter key. 

For more information on turning a link into a button, see: https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Roles/button_role " 

### Actual behavior

The element is announced by screen readers as a link.

### Steps to reproduce

1. Go to an individual order in the admin.
2. Inspect the "Add" button in the Add Note section.

### WordPress Environment

Verified in Version 10.3.0-dev (nightly)

### Isolating the problem

- [x] I have deactivated other plugins and confirmed this bug occurs when only WooCommerce plugin is active.
- [x] This bug happens with a default WordPress theme active, or [Storefront](https://woocommerce.com/products/storefront/).
- [x] I can reproduce this bug consistently using the steps above.

**Comment by tjcafferkey** [comment]
<img src="https://uploads.linear.app/f629fcde-6f9b-48e8-8af3-60eed8823bff/bd21743d-37b1-45ff-b2e1-2515ed137261/86a11d0c-e896-4efd-a6e6-2dc21bfdbe4a?signature=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJwYXRoIjoiL2Y2MjlmY2RlLTZmOWItNDhlOC04YWYzLTYwZWVkODgyM2JmZi9iZDIxNzQzZC0zN2IxLTQ1ZmYtYjJlMS0yNTE1ZWQxMzcyNjEvODZhMTFkMGMtZTg5Ni00ZWZkLWE2ZTYtMmRjMjFiZmRiZTRhIiwiaWF0IjoxNzU5NDA4ODk2LCJleHAiOjMzMzI5OTY4ODk2fQ.lrAwYWcxN-4RPmDstC0QyZsjekX-4niDI_cQHtyR3xM " alt="CleanShot 2025-10-02 at 13.39.08.png" width="1908" data-linear-height="517" />

Thanks for opening this issue! Unless I am misunderstanding this is correctly rendered as a button element? Also see the relevant code here: [https://github.com/woocommerce/woocommerce/blob/trunk/plugins/woocommerce/includes/admin/meta-boxes/class-wc-meta-box-order-notes.php#L50](https://github.com/woocommerce/woocommerce/blob/trunk/plugins/woocommerce/includes/admin/meta-boxes/class-wc-meta-box-order-notes.php#L50)

I will close this issue but feel free to reopen it if there's anything I have missed/misunderstood.

**Comment by amberhinds** [comment]
You're right - I'm seeing this as a button in Woo core or when I inspect the element as an administrator.  I think that this is actually an issue in the product vendors add-on. Here's what I see if inspecting that element as a vendor user:

<img width="1725" height="963" alt="Image" src="https://github.com/user-attachments/assets/9e3a3a5f-7b84-4017-912c-2c6fd2835a3c" />

Here's the system status report for the site where we're seeing this issue:

<details><summary>System Status Report</summary>

```
### WordPress Environment ###

WordPress address (URL): [Redacted]
Site address (URL): [Redacted]
WC Version: 10.3.0-dev
Legacy REST API Package Version: The Legacy REST API plugin is not installed on this site.
Action Scheduler Version: ✔ 3.9.3
Log Directory Writable: ✔
WP Version: ❌ 6.8.2 - There is a newer version of WordPress available (6.8.3)
WP Multisite: –
WP Memory Limit: 128 MB
WP Debug Mode: –
WP Cron: ✔
Language: en_US
External object cache: –

### Server Environment ###

Server Info: Apache/2.4.65 (Ubuntu) mod_fcgid/2.3.9 OpenSSL/3.0.13
Server Architecture: Linux 6.8.0-71-generic x86_64
PHP Version: 8.2.29
PHP Post Max Size: 256 MB
PHP Time Limit: 120
PHP Max Input Vars: 5000
cURL Version: 8.5.0
OpenSSL/3.0.13

SUHOSIN Installed: –
MySQL Version: 10.11.13-MariaDB-ubu2004
Max Upload Size: 256 MB
Default Timezone is UTC: ✔
fsockopen/cURL: ✔
SoapClient: ✔
DOMDocument: ✔
GZip: ✔
Multibyte String: ✔
Remote Post: ✔
Remote Get: ✔

### Database ###

[REDACTED]

### Post Type Counts ###

attachment: 11
nav_menu_item: 5
page: 11
post: 2
product: 23
product_variation: 61
revision: 6
shop_order: 40
wp_font_face: 36
wp_font_family: 12
wp_navigation: 1

### Security ###

Secure connection (HTTPS): ✔
Hide errors from visitors: ✔

### Active Plugins (13) ###

Accessibility Checker: by Equalize Digital – 1.33.0
AutomateWoo - Birthdays Add-on: by WooCommerce – 1.3.52
AutomateWoo - Refer A Friend Add-on: by WooCommerce – 2.8.10
AutomateWoo: by WooCommerce – 6.1.17
Classic Editor: by WordPress Contributors – 1.6.7
Pinterest for WooCommerce: by WooCommerce – 1.4.21
User Switching: by John Blackbourn – 1.10.0
WooCommerce Smooth Generator: by Automattic – 1.2.2
WooCommerce.com Update Manager: by Automattic – 1.0.3
WooCommerce Payfast Gateway: by WooCommerce – 1.7.3
WooCommerce Product Vendors: by WooCommerce – 2.4.7
WooCommerce: by Automattic – 10.3.0-dev
WP Mail Logging: by WP Mail Logging Team – 1.15.0

### Inactive Plugins (0) ###


### Settings ###

Legacy API Enabled: –
Force SSL: –
Currency: ZAR (R)
Currency Position: left
Thousand Separator: 
Decimal Separator: ,
Number of Decimals: 2
Taxonomies: Product Types: external (external)
grouped (grouped)
simple (simple)
variable (variable)

Taxonomies: Product Visibility: exclude-from-catalog (exclude-from-catalog)
exclude-from-search (exclude-from-search)
featured (featured)
outofstock (outofstock)
rated-1 (rated-1)
rated-2 (rated-2)
rated-3 (rated-3)
rated-4 (rated-4)
rated-5 (rated-5)

Connected to WooCommerce.com: ✔
Enforce Approved Product Download Directories: ✔
HPOS feature enabled: –
Order datastore: WC_Order_Data_Store_CPT
HPOS data sync enabled: –
Enabled Features: analytics
marketplace
order_attribution
site_visibility_badge
remote_logging
email_improvements
blueprint
point_of_sale


### Logging ###

Enabled: ✔
Handler: Automattic\WooCommerce\Internal\Admin\Logging\LogHandlerFileV2
Retention period: 30 days
Level threshold: –
Log directory size: 11 KB

### WC Pages ###

Shop base: #6 - /shop/
Cart: #7 - /cart/ -  Contains the woocommerce/cart block
Checkout: #8 - /checkout/ -  Contains the woocommerce/checkout block
My account: #9 - /my-account/ -  Contains the [woocommerce_my_account] shortcode
Terms and conditions: ❌ Page not set

### Theme ###

Name: Storefront
Version: 4.6.1
Author URL: https://woocommerce.com/
Child Theme: ❌ – If you are modifying WooCommerce on a parent theme that you did not build personally we recommend using a child theme. See: How to create a child theme
Theme type: Classic theme
WooCommerce Support: ✔

### Templates ###

Overrides: –

### Admin ###

Enabled Features: activity-panels
analytics
product-block-editor
coupons
core-profiler
customize-store
customer-effort-score-tracks
import-products-task
experimental-fashion-sample-products
shipping-smart-defaults
shipping-setting-tour
homescreen
marketing
mobile-app-banner
onboarding
onboarding-tasks
pattern-toolkit-full-composability
product-custom-fields
remote-inbox-notifications
remote-free-extensions
payment-gateway-suggestions
printful
shipping-label-banner
subscriptions
store-alerts
transient-notices
woo-mobile-welcome
wc-pay-promotion
wc-pay-welcome-page
launch-your-store

Disabled Features: product-data-views
experimental-blocks
experimental-iapi-mini-cart
experimental-iapi-runtime
coming-soon-newsletter-template
minified-js
product-pre-publish-modal
settings
async-product-editor-category-field
product-editor-template-system
use-wp-horizon
rest-api-v4

Daily Cron: ❌ Not scheduled
Options: ✔
Notes: 77
Onboarding: completed

### Action Scheduler ###

Canceled: 2
Oldest: 2025-10-06 19:00:30 +0000
Newest: 2025-10-07 19:18:36 +0000

Complete: 570
Oldest: 2025-09-14 19:06:36 +0000
Newest: 2025-10-07 19:24:35 +0000

Failed: 5
Oldest: 2025-08-15 15:46:04 +0000
Newest: 2025-09-30 20:50:22 +0000

Pending: 23
Oldest: 2025-10-07 19:24:54 +0000
Newest: 2025-10-15 10:17:08 +0000


### Status report information ###

Generated at: 2025-10-07 19:24:52 +00:00
```
</details>

@vikrampm1 - this was previously marked as a Woo core issue in the audit report, but it seems that was incorrect.
